sktime/forecasting/compose/_pipeline.py

```diff
@@ -1448,3 +1450,192 @@ def get_test_params(cls, parameter_set="default"):
+
+            # check that all permutation are str values
+            if not all(isinstance(item, str) for item in permutation):
+                raise ValueError("permutation must be None or a list of strings")
+
+            # check that permutation contains same step names as given in steps
+            if not set(estimator_dict.keys()) == set(permutation):
+                raise ValueError(
+                    f"""Permutations must contain exactly the same step names as
+                    the names of steps in getattr(estimator, steps_arg),
+                    found tuple names {set(estimator_dict.keys())} but got
+                    permutation {set(permutation)}."""
+                )
+
+            estimator_tuples_permuted = [(k, estimator_dict[k]) for k in permutation]
+
+            self.estimator_ = estimator.clone()
+            self.estimator_.set_params(**{steps_arg: estimator_tuples_permuted})
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        from sktime.forecasting.naive import NaiveForecaster
+        from sktime.transformations.series.boxcox import BoxCoxTransformer
+        from sktime.transformations.series.exponent import ExponentTransformer
+
+        # transformers mixed with-without fit, ForecastingPipeline
+        # steps are (str, estimator)
+        params1 = {
+            "estimator": ForecastingPipeline(
+                [
+                    ("foo", BoxCoxTransformer()),
+                    ("bar", ExponentTransformer(3)),
+                    ("foobar", NaiveForecaster()),
+                ]
+            ),
+            "permutation": ["bar", "foo", "foobar"],
+        }
+
+        # transformers have no fit, TransformedTargetForecaster
+        # steps are only estimator
+        params2 = {
+            "estimator": TransformedTargetForecaster(
+                [ExponentTransformer(0.5), NaiveForecaster(), ExponentTransformer(3)]
+            ),
+            "permutation": [
+                "NaiveForecaster",
+                "ExponentTransformer_1",
+                "ExponentTransformer_2",
+            ],
+        }
```

Comment:
Request to add more tests around tuning of permute, and tuning of other parameters other than the steps order when the pipeline is wrapped in `Permute`.

Reply:
Sure, I agree that it should work ok.  My main concern with requesting this (and actually perhaps it is better solved by clear documentation rather than a test)  is - before we might have been able to tune pipeline's parameter foo by telling gscv to tune 'foo' but now we will need to tell it to tune 'pipeline.foo' as pipeline is now wrapped in another layer.  

I don't think that this is broken per say - but I do anticipate it being confusing, and thus worth pointing out somewhere.  Documentation would be fine.

Reply:
hm, what test specifically would you like to add? A grid search where we tune `pipeline__foo`?

Reply:
I think there is now a example in doctring section `Examples`, so would be fine for me. This is also executed when running tests.